pyatlan/pkg/utils.py

```diff
@@ -117,3 +167,54 @@ def validate_connector_and_connection(v):
     from pyatlan.pkg.models import ConnectorAndConnection
 
     return parse_raw_as(ConnectorAndConnection, v)
+
+
+def has_handler(logger: logging.Logger, handler_class) -> bool:
+    """
+    Checks if a logger or its ancestor has a handler of a specific class. The function
+    iterates through the logger's handlers and optionally ascends the logger hierarchy,
+    checking each logger's handlers for an instance of the specified handler class.
+
+    Args:
+        logger (logging.Logger): The logger to inspect for the handler.
+        handler_class: The class of the handler to look for.
+
+    Returns:
+        bool: True if the handler of the specified class is found, False otherwise.
+    """
+    c: Optional[logging.Logger] = logger
+    while c:
+        for hdlr in c.handlers:
+            if isinstance(hdlr, handler_class):
+                return True
+        c = c.parent if c.propagate else None
+    return False
+
+
+def add_otel_handler(
+    logger: logging.Logger, level: Union[int, str], resource: dict
+) -> None:
+    """
+    Adds an OpenTelemetry handler to the logger if not already present.
+
+    Args:
+        logger (logging.Logger): The logger to which the handler will be added.
+        level (int | str): The logging level.
+        resource (dict): A dictionary of resource attributes to be associated with the logger.
+    """
+    if OTEL_IMPORTS_AVAILABLE and not has_handler(logger, LoggingHandler):
+        workflow_node_name = os.getenv("OTEL_WF_NODE_NAME", "")
+        if workflow_node_name:
+            resource["k8s.workflow.node.name"] = workflow_node_name
+        logger_provider = LoggerProvider(Resource.create(resource))
+        otel_log_exporter = OTLPLogExporter(
+            endpoint=os.getenv("OTEL_EXPORTER_OTLP_ENDPOINT"), insecure=True
+        )
+        logger_provider.add_log_record_processor(
+            CustomBatchLogRecordProcessor(otel_log_exporter)
+        )
+
+        otel_handler = LoggingHandler(level=level, logger_provider=logger_provider)
+        otel_handler.setLevel(level)
+        logger.addHandler(otel_handler)
+        logger.info("OpenTelemetry handler added to the logger.")
```

Comment:
Do we need to update `Renderers.pkl` in the `Java SDK` as well (especially to include these new methods in the Python imports), since we’ll be using the `.pkl-based` toolkit to render the custom package? 🤔 

For example: https://github.com/atlanhq/atlan-java/blob/5c116422f231ebb1efa99bcfaf053100f72c2d2c/package-toolkit/config/src/main/resources/Renderers.pkl#L366

Reply:
Yes I will update those to include the requisite imports in the requirements.txt and the invocation of `add_otel_handler` in the code.